src/Infrastructure/Event/CliReceiverSetupListener.php

```diff
@@ -7,33 +7,149 @@
 
 namespace SprykerSdk\Sdk\Infrastructure\Event;
 
+use SprykerSdk\Sdk\Infrastructure\Service\FilesystemInitInterface;
+use SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\ApiInteractionProcessor;
+use SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\CliInteractionProcessor;
+use SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\InteractionProcessorReceiverInterface;
 use Symfony\Component\Console\Event\ConsoleCommandEvent;
+use Symfony\Component\Console\Helper\DebugFormatterHelper;
+use Symfony\Component\Console\Helper\FormatterHelper;
+use Symfony\Component\Console\Helper\HelperSet;
+use Symfony\Component\Console\Helper\ProcessHelper;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Output\BufferedOutput;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
 
 class CliReceiverSetupListener
 {
     /**
-     * @var iterable<\SprykerSdk\Sdk\Infrastructure\Event\InputOutputReceiverInterface>
+     * @var iterable<\SprykerSdk\Sdk\Infrastructure\Event\ReceiverInterface>
      */
     protected iterable $inputOutputConnectors;
 
     /**
-     * @param iterable<\SprykerSdk\Sdk\Infrastructure\Event\InputOutputReceiverInterface> $inputOutputConnectors
+     * @var \SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\ApiInteractionProcessor
      */
-    public function __construct(iterable $inputOutputConnectors)
-    {
+    protected ApiInteractionProcessor $apiInteractionProcessor;
+
+    /**
+     * @var \SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\CliInteractionProcessor
+     */
+    protected CliInteractionProcessor $cliInteractionProcessor;
+
+    /**
+     * @var string
+     */
+    protected string $sdkPath;
+
+    /**
+     * @param iterable $inputOutputConnectors
+     * @param \SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\ApiInteractionProcessor $apiInteractionProcessor
+     * @param \SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\CliInteractionProcessor $cliInteractionProcessor
+     * @param string $sdkPath
+     */
+    public function __construct(
+        iterable $inputOutputConnectors,
+        ApiInteractionProcessor $apiInteractionProcessor,
+        CliInteractionProcessor $cliInteractionProcessor,
+        string $sdkPath
+    ) {
         $this->inputOutputConnectors = $inputOutputConnectors;
+        $this->apiInteractionProcessor = $apiInteractionProcessor;
+        $this->cliInteractionProcessor = $cliInteractionProcessor;
+        $this->sdkPath = $sdkPath;
     }
 
     /**
      * @param \Symfony\Component\Console\Event\ConsoleCommandEvent $event
      *
      * @return void
      */
-    public function beforeConsoleCommand(ConsoleCommandEvent $event)
+    public function beforeConsoleCommand(ConsoleCommandEvent $event): void
     {
+        if (!$event->getCommand()) {
+            return;
+        }
+
         foreach ($this->inputOutputConnectors as $inputOutputConnector) {
```

Comment:
I'd replace those interfaces with the factories of the target objects

Reply:
Not needed now.  Do not complicate simple logic